lxd/instance/drivers/driver_qemu.go

```diff
@@ -7318,7 +7318,14 @@ func (d *qemu) vsockID() int {
 	// We then add the VM's own instance id (1 or higher) to give us a
 	// unique, non-clashing context ID for our guest.
 
-	return int(d.state.OS.VsockID) + 1 + d.id
+	vsockID, err := vsock.ContextID()
```

Comment:
This should be extracted from the features detected at LXD start time, see `info := DriverStatuses()[instancetype.VM].Info` usage.

Reply:
Otherwise we are duplicating logic.